crates/topos-tce-broadcast/src/double_echo/mod.rs

```diff
@@ -25,6 +27,12 @@ pub struct DoubleEcho {
     task_manager_message_sender: mpsc::Sender<DoubleEchoCommand>,
     /// The overview of the network, which holds echo and ready subscriptions and the network size
     pub subscriptions: SubscriptionsView,
+    /// Public ETH address
+    pub validator_id: ValidatorId,
+    /// Keypair to sign and verify ECHO and READY messages
+    pub wallet: LocalWallet,
```

Comment:
Random question: the `Wallet` contains the address already; are we checking that it is the same one as the `ValidatorId`? Could we somehow use the wallet address instead of having to use our own field everywhere?

Reply:
I think the idea was to be chain agnostic. So we might not always have Ethereum public addresses? But yeah, totally see that reasoning. @hadjiszs might have some thoughts on this as well.

Reply:
It seems like we're assuming that addresses are ethereum style `H160` most places anyway. And we assume EVM compatibility in many places yeah?

Reply:
Yeah we do. The idea is that at some point we don't need to rely on Ethereum and can plug in non-Blockchain networks as well. But yeah, design decision. I  can see pros and cons on both sides.

Reply:
I open an issue so we can discuss this in another PR?